### PR TITLE
Simplified the Lexer

### PR DESCRIPTION
**Pull request by sgraf812**
Got rid of some redundant state. Also fixed the git hook to cope with renames.

### CODE REVIEW

**Comment by sgraf812** [comment]
Uhmm.. that probably was one removed redundancy too much...

File: src/main/java/minijava/lexer/Lexer.java [hunk 1]
```diff
@@ -409,34 +406,24 @@ private Token createToken(Terminal terminal, String content) {
...
-    } else {
-      current = parseNextToken();
-    }
-    return current;
-  }
-
   @Override
   public boolean hasNext() {
-    return numberOfEOFs < 1;
+    return token == null || !token.isEOF();
```
Comment (sherter) [inline src/main/java/minijava/lexer/Lexer.java:411]: `token == null` is only true once (in the beginning). _**If**_ we were optimizing for performance, the test for `!token.isEOF()` should probably be the first one :stuck_out_tongue:

Comment (sgraf812) [reply src/main/java/minijava/lexer/Lexer.java:411]: Even if, I'm inclined to say this won't matter.

Comment (sherter) [reply src/main/java/minijava/lexer/Lexer.java:411]: agreed :wink:

**Pull request merged**